pennylane/workflow/execution.py

```diff
@@ -583,6 +583,8 @@ def cost_fn(params, x):
         if config.mcm_config.postselect_mode == "hw-like":
             raise ValueError("Using postselect_mode='hw-like' is not supported with jax-jit.")
         config.mcm_config.postselect_mode = "fill-shots"
+    if config.mcm_config.mcm_method == "single-branch-statistics" and not qml.compiler.active():
+        raise ValueError("Cannot use mcm_method='single-branch-statistics' without qml.qjit")

```

Comment:
I'm wondering if I even need to check if there is an active compiler or not here, since `qml.execute` shouldn't be getting hit when qjit-ing. Also would like thoughts on if it makes more sense for this check to be in `QNode.__call__` rather than `qml.execute`.

Reply:
Agreed we would never invoke execute. QNode.__call__ would work as well, although we don't invoke that function either.

Reply:
I think I will move the check to `QNode.__call__`. CodeFactor is giving a "Very Complex Method" warning, and it seems like removing a mere two lines from `qml.execute` might fix that 🙃

Reply:
I'm OK with either, but in favor of reducing the complexity of `execute`.

Reply:
I added it to `QNode.__call__`.